compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -783,15 +783,7 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
                 }
             }
             Abi::ScalarPair(a_layout, b_layout) => {
-                // There is no `rustc_layout_scalar_valid_range_start` for pairs, so
-                // we would validate these things as we descend into the fields,
-                // but that can miss bugs in layout computation. Layout computation
-                // is subtle due to enums having ScalarPair layout, where one field
-                // is the discriminant.
-                if cfg!(debug_assertions)
-                    && !a_layout.is_uninit_valid()
-                    && !b_layout.is_uninit_valid()
-                {
+                if !a_layout.is_uninit_valid() && !b_layout.is_uninit_valid() {
                     // We can only proceed if *both* scalars need to be initialized.
                     // FIXME: find a way to also check ScalarPair when one side can be uninit but
                     // the other must be init.
```

Comment:
This comment should probably be above the `if`

Reply:
Also the condition is wrong, but that one I am unable to test I believe.

Reply:
Wrong in which sense? We of course should also check if only one field requires init, but as the FIXME says we have no good way of doing that.